### PR TITLE
Add additional VVC program suits into autobuild media

### PR DESCRIPTION
**Pull request by MartinEesmaa**
See issue: #1810 

Added vvenc & vvdec, but not tested yet, others vvc programs are coming soon...

Update of 16th June 2022: vvenc & vvdec is only now support of media autobuild suite, others were not.

- [X] vvenc & vvdec
- [ ] Bitmovin VVDec Player
- [ ] FFmpeg (libvvdec plugin)
- [ ] VLC InterDigital Plugin?

- Martin Eesmaa

### CODE REVIEW

**Comment by MartinEesmaa** [comment]
vvenc is now tested and success, but a little bug. It's easy to fix the bug, no problem at all:
```
17:43:26   Running git clone for vvenc...
17:43:38 ┌ vvenc git  ................................ [Recently updated]
17:43:38 ├ Running cmake...
17:43:44 ├ Running build...
17:46:43 ├ Running install...
17:46:44 ├ /local64/bin-video/vvenc/vvencinterfacetest.exe  . [Not found]
17:46:44 └ vvenc-git  .......................................... [Failed]

Try deleting '/build/vvenc-git' and start the script again.
If you are sure there are no dependencies, <Enter> to continue building.
Close this window if you wish to stop building.
```

- Martin Eesmaa

File: build/media-suite_compile.sh [hunk 1]
```diff
@@ -1823,6 +1825,24 @@ if [[ $bits = 64bit && $uvg266 = y ]] &&
...
+_check=(bin-video/vvenc/vvenc{app,FFapp}.exe)
+if [[ $bits = 64bit && $vvenc = y ]] &&
+    do_vcs "https://github.com/fraunhoferhhi/vvenc.git" vvenc; then
+    do_uninstall bin-video/vvenc
+    do_cmakeinstall -DCMAKE_INSTALL_BINDIR="$LOCALDESTDIR"/bin-video/vvenc \
+        -DBUILD_STATIC=on
+    do_checkIfExist
+fi
+
+_check=(bin-video/vvdec/vvdecapp.exe)
```
Comment (1480c1) [inline build/media-suite_compile.sh:1837]:
```suggestion
_check=(bin-video/vvdecapp.exe
    vvdec/vvdec.h
    libvvdec.{a,pc}
    lib/cmake/vvdec/vvdecConfig.cmake)
```

similar issue here

File: build/media-suite_compile.sh [hunk 2]
```diff
@@ -1823,6 +1825,24 @@ if [[ $bits = 64bit && $uvg266 = y ]] &&
     do_checkIfExist
 fi
 
+_check=(bin-video/vvenc/vvenc{app,FFapp}.exe)
+if [[ $bits = 64bit && $vvenc = y ]] &&
+    do_vcs "https://github.com/fraunhoferhhi/vvenc.git" vvenc; then
```
Comment (1480c1) [inline build/media-suite_compile.sh:1830]:
```suggestion
    do_vcs "https://github.com/fraunhoferhhi/vvenc.git"; then
```
the third argument is unnecessary as git can infer `vvenc.git` goes to `vvenc`, it's mainly for repos that have weird names and such

File: build/media-suite_compile.sh [hunk 3]
```diff
@@ -1823,6 +1825,24 @@ if [[ $bits = 64bit && $uvg266 = y ]] &&
...
+    do_uninstall bin-video/vvenc
+    do_cmakeinstall -DCMAKE_INSTALL_BINDIR="$LOCALDESTDIR"/bin-video/vvenc \
+        -DBUILD_STATIC=on
+    do_checkIfExist
+fi
+
+_check=(bin-video/vvdec/vvdecapp.exe)
+if [[ $bits = 64bit && $vvdec = y ]] &&
+    do_vcs "https://github.com/fraunhoferhhi/vvdec.git" vvdec; then
```
Comment (1480c1) [inline build/media-suite_compile.sh:1839]:
```suggestion
    do_vcs "https://github.com/fraunhoferhhi/vvdec.git"; then
```

same here

File: build/media-suite_compile.sh [hunk 4]
```diff
@@ -1823,6 +1825,24 @@ if [[ $bits = 64bit && $uvg266 = y ]] &&
...
+    do_uninstall bin-video/vvdec
```
Comment (1480c1) [inline build/media-suite_compile.sh:1840]:
```suggestion
    do_uninstall include/vvdec lib/cmake/vvdec "${_check[@]}"
```

File: build/media-suite_compile.sh [hunk 5]
```diff
@@ -1823,6 +1825,24 @@ if [[ $bits = 64bit && $uvg266 = y ]] &&
...
+    do_cmakeinstall -DCMAKE_INSTALL_BINDIR="$LOCALDESTDIR"/bin-video/vvdec \
```
Comment (1480c1) [inline build/media-suite_compile.sh:1841]:
```suggestion
    do_cmakeinstall video \
```

**Review by 1480c1 — COMMENTED** [review]
Please also add these entries inside the README.md near uvg266 for now

**Comment by MartinEesmaa** [comment]
Let me check, git pull and see how works on my virtual machine. I'll test it out...

**Comment by MartinEesmaa** [comment]
Hmm.. it sound problem with no founded :/
virtualboxmartineesmaamastest

- Martin Eesmaa

**Comment by MartinEesmaa** [comment]
Same... so the solution is to add app before .exe in compiler shell
VirtualBox_Windows Server 2019_14_06_2022_18_13_21

**Comment by MartinEesmaa** [comment]
Report commit 25b6f2e log:
```
18:47:27   Running git update for vvenc...
18:47:28 ┌ vvenc git  ................................ [Recently updated]
18:47:28 ├ Running cmake...
18:47:31 ├ Running build...
18:50:15 ├ Running install...
18:50:15 ├ Stripping...
18:50:16 └ vvenc-git  ......................................... [Updated]
18:50:17   Running git update for vvdec...
18:50:23 ┌ vvenc git  ................................ [Recently updated]
18:50:24 ├ Running cmake...
18:50:31 ├ Running build...
18:52:22 ├ Running install...
18:52:22 ├ Stripping...
18:52:22 └ vvenc-git  ......................................... [Updated]
18:52:22 Finished 64bit compilation of all tools
```

**Comment by 1480c1** [comment]
lgtm, do you want me to cleanup the commits, rebase and leave them as is, or squash them into just one commit?

**Comment by MartinEesmaa** [comment]
@1480c1, might be just leave this commits unless I tell it's ready to merge. I tried compile of [my forked repository](https://github.com/MartinEesmaa/FFmpegVVC-latest) + vvc commits, but it did not work, so tbiat/FFmpeg works only modified version without latest commits of 2022.

```
marti@LAPTOP-HM70RGB0 MINGW64 ~/FFmpegVVC-latest
$ make -j
CC      libavdevice/version.o
CC      libavfilter/version.o
CC      libavformat/rawenc.o
CC      libavformat/version.o
libavformat/rawenc.c:424:26: warning: initialization of 'int (*)(struct AVFormatContext *, struct AVStream *, const AVPacket *)' from incompatible pointer type 'int (*)(struct AVFormatContext *, const AVPacket *)' [-Wincompatible-pointer-types]
  424 |     .check_bitstream   = vvc_check_bitstream,
      |                          ^~~~~~~~~~~~~~~~~~~
libavformat/rawenc.c:424:26: note: (near initialization for 'ff_vvc_muxer.check_bitstream')
CC      libavcodec/libvvdec.o
CC      libavcodec/libvvenc.o
CC      libavcodec/mjpegdec.o
CC      libavcodec/mjpegdec_common.o
CC      libavcodec/mjpegenc.o
CC      libavcodec/mjpegenc_common.o
libavcodec/libvvdec.c:399:1: error: unknown type name 'FFCodec'; did you mean 'AVCodec'?
  399 | FFCodec ff_libvvdec_decoder = {
      | ^~~~~~~
      | AVCodec
libavcodec/libvvdec.c:400:3: error: field name not in record or union initializer
  400 |   .p.name            = "libvvdec",
      |   ^
libavcodec/libvvdec.c:400:3: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:400:24: warning: initialization of 'int' from 'const char *' makes integer from pointer without a cast [-Wint-conversion]
  400 |   .p.name            = "libvvdec",
      |                        ^~~~~~~~~~
libavcodec/libvvdec.c:400:24: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:400:24: error: initializer element is not computable at load time
libavcodec/libvvdec.c:400:24: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:401:3: error: field name not in record or union initializer
  401 |   .p.long_name       = "H.266 / VVC Decoder VVdeC",
      |   ^
libavcodec/libvvdec.c:401:3: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:401:24: warning: excess elements in scalar initializer
  401 |   .p.long_name       = "H.266 / VVC Decoder VVdeC",
      |                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~
libavcodec/libvvdec.c:401:24: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:402:3: error: field name not in record or union initializer
  402 |   .p.type            = AVMEDIA_TYPE_VIDEO,
      |   ^
libavcodec/libvvdec.c:402:3: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:402:24: warning: excess elements in scalar initializer
  402 |   .p.type            = AVMEDIA_TYPE_VIDEO,
      |                        ^~~~~~~~~~~~~~~~~~
libavcodec/libvvdec.c:402:24: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:403:3: error: field name not in record or union initializer
  403 |   .p.id              = AV_CODEC_ID_VVC,
      |   ^
libavcodec/libvvdec.c:403:3: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:403:24: warning: excess elements in scalar initializer
  403 |   .p.id              = AV_CODEC_ID_VVC,
      |                        ^~~~~~~~~~~~~~~
libavcodec/libvvdec.c:403:24: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:404:3: error: field name not in record or union initializer
  404 |   .priv_data_size    = sizeof(VVdeCContext),
      |   ^
libavcodec/libvvdec.c:404:3: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:404:24: warning: excess elements in scalar initializer
  404 |   .priv_data_size    = sizeof(VVdeCContext),
      |                        ^~~~~~
libavcodec/libvvdec.c:404:24: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:405:3: error: field name not in record or union initializer
  405 |   .init              = ff_vvdec_decode_init,
      |   ^
libavcodec/libvvdec.c:405:3: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:405:24: warning: excess elements in scalar initializer
  405 |   .init              = ff_vvdec_decode_init,
      |                        ^~~~~~~~~~~~~~~~~~~~
libavcodec/libvvdec.c:405:24: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:406:3: error: field name not in record or union initializer
  406 |   .decode            = ff_vvdec_decode_frame,
      |   ^
libavcodec/libvvdec.c:406:3: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:406:24: warning: excess elements in scalar initializer
  406 |   .decode            = ff_vvdec_decode_frame,
      |                        ^~~~~~~~~~~~~~~~~~~~~
libavcodec/libvvdec.c:406:24: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:407:3: error: field name not in record or union initializer
  407 |   .close             = ff_vvdec_decode_close,
      |   ^
libavcodec/libvvdec.c:407:3: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:407:24: warning: excess elements in scalar initializer
  407 |   .close             = ff_vvdec_decode_close,
      |                        ^~~~~~~~~~~~~~~~~~~~~
libavcodec/libvvdec.c:407:24: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:408:3: error: field name not in record or union initializer
  408 |   .flush             = ff_vvdec_decode_flush,
      |   ^
libavcodec/libvvdec.c:408:3: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:408:24: warning: excess elements in scalar initializer
  408 |   .flush             = ff_vvdec_decode_flush,
      |                        ^~~~~~~~~~~~~~~~~~~~~
libavcodec/libvvdec.c:408:24: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:409:3: error: field name not in record or union initializer
  409 |   .p.capabilities    = AV_CODEC_CAP_DELAY | AV_CODEC_CAP_OTHER_THREADS,
      |   ^
libavcodec/libvvdec.c:409:3: note: (near initialization for 'ff_libvvdec_decoder')
In file included from libavcodec/avcodec.h:40,
                 from libavcodec/libvvdec.c:27:
libavcodec/codec.h:82:42: warning: excess elements in scalar initializer
   82 | #define AV_CODEC_CAP_DELAY               (1 <<  5)
      |                                          ^
libavcodec/libvvdec.c:409:24: note: in expansion of macro 'AV_CODEC_CAP_DELAY'
  409 |   .p.capabilities    = AV_CODEC_CAP_DELAY | AV_CODEC_CAP_OTHER_THREADS,
      |                        ^~~~~~~~~~~~~~~~~~
libavcodec/codec.h:82:42: note: (near initialization for 'ff_libvvdec_decoder')
   82 | #define AV_CODEC_CAP_DELAY               (1 <<  5)
      |                                          ^
libavcodec/libvvdec.c:409:24: note: in expansion of macro 'AV_CODEC_CAP_DELAY'
  409 |   .p.capabilities    = AV_CODEC_CAP_DELAY | AV_CODEC_CAP_OTHER_THREADS,
      |                        ^~~~~~~~~~~~~~~~~~
libavcodec/libvvdec.c:410:3: error: field name not in record or union initializer
  410 |   .bsfs              = "vvc_mp4toannexb",
      |   ^
libavcodec/libvvdec.c:410:3: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:410:24: warning: excess elements in scalar initializer
  410 |   .bsfs              = "vvc_mp4toannexb",
      |                        ^~~~~~~~~~~~~~~~~
libavcodec/libvvdec.c:410:24: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:411:3: error: field name not in record or union initializer
  411 |   .caps_internal     = FF_CODEC_CAP_AUTO_THREADS,
      |   ^
libavcodec/libvvdec.c:411:3: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:411:24: error: 'FF_CODEC_CAP_AUTO_THREADS' undeclared here (not in a function); did you mean 'AV_CODEC_CAP_AUTO_THREADS'?
  411 |   .caps_internal     = FF_CODEC_CAP_AUTO_THREADS,
      |                        ^~~~~~~~~~~~~~~~~~~~~~~~~
      |                        AV_CODEC_CAP_AUTO_THREADS
libavcodec/libvvdec.c:411:24: warning: excess elements in scalar initializer
libavcodec/libvvdec.c:411:24: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:412:3: error: field name not in record or union initializer
  412 |   .p.pix_fmts        = pix_fmts_vvc,
      |   ^
libavcodec/libvvdec.c:412:3: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:412:24: warning: excess elements in scalar initializer
  412 |   .p.pix_fmts        = pix_fmts_vvc,
      |                        ^~~~~~~~~~~~
libavcodec/libvvdec.c:412:24: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:413:3: error: field name not in record or union initializer
  413 |   .p.priv_class      = &libvvdec_class,
      |   ^
libavcodec/libvvdec.c:413:3: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:413:24: warning: excess elements in scalar initializer
  413 |   .p.priv_class      = &libvvdec_class,
      |                        ^
libavcodec/libvvdec.c:413:24: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:414:3: error: field name not in record or union initializer
  414 |   .p.wrapper_name      = "libvvdec",
      |   ^
libavcodec/libvvdec.c:414:3: note: (near initialization for 'ff_libvvdec_decoder')
libavcodec/libvvdec.c:414:26: warning: excess elements in scalar initializer
  414 |   .p.wrapper_name      = "libvvdec",
      |                          ^~~~~~~~~~
libavcodec/libvvdec.c:414:26: note: (near initialization for 'ff_libvvdec_decoder')
CC      libavcodec/mjpegenc_huffman.o
CC      libavcodec/mlp.o
make: *** [ffbuild/common.mak:81: libavcodec/libvvdec.o] Error 1
make: *** Waiting for unfinished jobs....
CC      libavcodec/mlp_parse.o
CC      libavcodec/mlp_parser.o
libavcodec/libvvenc.c: In function 'ff_vvenc_encode_frame':
libavcodec/libvvenc.c:131:9: error: implicit declaration of function 'ff_alloc_packet2'; did you mean 'ff_alloc_packet'? [-Werror=implicit-function-declaration]
  131 |         ff_alloc_packet2(avctx, avpkt, pktSize, 0);
      |         ^~~~~~~~~~~~~~~~
      |         ff_alloc_packet
libavcodec/libvvenc.c: At top level:
libavcodec/libvvenc.c:162:1: error: unknown type name 'FFCodec'; did you mean 'AVCodec'?
  162 | FFCodec ff_libvvenc_encoder = {
      | ^~~~~~~
      | AVCodec
libavcodec/libvvenc.c:163:5: error: field name not in record or union initializer
  163 |     .p.name           = "libvvenc",
      |     ^
libavcodec/libvvenc.c:163:5: note: (near initialization for 'ff_libvvenc_encoder')
libavcodec/libvvenc.c:163:25: warning: initialization of 'int' from 'const char *' makes integer from pointer without a cast [-Wint-conversion]
  163 |     .p.name           = "libvvenc",
      |                         ^~~~~~~~~~
libavcodec/libvvenc.c:163:25: note: (near initialization for 'ff_libvvenc_encoder')
libavcodec/libvvenc.c:163:25: error: initializer element is not computable at load time
libavcodec/libvvenc.c:163:25: note: (near initialization for 'ff_libvvenc_encoder')
libavcodec/libvvenc.c:164:5: error: field name not in record or union initializer
  164 |     .p.long_name      = "H.266 / VVC Encoder VVenC",
      |     ^
libavcodec/libvvenc.c:164:5: note: (near initialization for 'ff_libvvenc_encoder')
libavcodec/libvvenc.c:164:25: warning: excess elements in scalar initializer
  164 |     .p.long_name      = "H.266 / VVC Encoder VVenC",
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
libavcodec/libvvenc.c:164:25: note: (near initialization for 'ff_libvvenc_encoder')
libavcodec/libvvenc.c:165:5: error: field name not in record or union initializer
  165 |     .p.type           = AVMEDIA_TYPE_VIDEO,
      |     ^
libavcodec/libvvenc.c:165:5: note: (near initialization for 'ff_libvvenc_encoder')
libavcodec/libvvenc.c:165:25: warning: excess elements in scalar initializer
  165 |     .p.type           = AVMEDIA_TYPE_VIDEO,
      |                         ^~~~~~~~~~~~~~~~~~
libavcodec/libvvenc.c:165:25: note: (near initialization for 'ff_libvvenc_encoder')
libavcodec/libvvenc.c:166:5: error: field name not in record or union initializer
  166 |     .p.id             = AV_CODEC_ID_VVC,
      |     ^
libavcodec/libvvenc.c:166:5: note: (near initialization for 'ff_libvvenc_encoder')
libavcodec/libvvenc.c:166:25: warning: excess elements in scalar initializer
  166 |     .p.id             = AV_CODEC_ID_VVC,
      |                         ^~~~~~~~~~~~~~~
libavcodec/libvvenc.c:166:25: note: (near initialization for 'ff_libvvenc_encoder')
libavcodec/libvvenc.c:167:5: error: field name not in record or union initializer
  167 |     .priv_data_size   = sizeof(VVEnCContext),
      |     ^
libavcodec/libvvenc.c:167:5: note: (near initialization for 'ff_libvvenc_encoder')
libavcodec/libvvenc.c:167:25: warning: excess elements in scalar initializer
  167 |     .priv_data_size   = sizeof(VVEnCContext),
      |                         ^~~~~~
libavcodec/libvvenc.c:167:25: note: (near initialization for 'ff_libvvenc_encoder')
libavcodec/libvvenc.c:168:5: error: field name not in record or union initializer
  168 |     .init             = ff_vvenc_encode_init,
      |     ^
libavcodec/libvvenc.c:168:5: note: (near initialization for 'ff_libvvenc_encoder')
libavcodec/libvvenc.c:168:25: warning: excess elements in scalar initializer
  168 |     .init             = ff_vvenc_encode_init,
      |                         ^~~~~~~~~~~~~~~~~~~~
libavcodec/libvvenc.c:168:25: note: (near initialization for 'ff_libvvenc_encoder')
libavcodec/libvvenc.c:169:5: error: field name not in record or union initializer
  169 |     .encode2          = ff_vvenc_encode_frame,
      |     ^
libavcodec/libvvenc.c:169:5: note: (near initialization for 'ff_libvvenc_encoder')
libavcodec/libvvenc.c:169:25: warning: excess elements in scalar initializer
  169 |     .encode2          = ff_vvenc_encode_frame,
      |                         ^~~~~~~~~~~~~~~~~~~~~
libavcodec/libvvenc.c:169:25: note: (near initialization for 'ff_libvvenc_encoder')
libavcodec/libvvenc.c:170:5: error: field name not in record or union initializer
  170 |     .close            = ff_vvenc_encode_close,
      |     ^
libavcodec/libvvenc.c:170:5: note: (near initialization for 'ff_libvvenc_encoder')
libavcodec/libvvenc.c:170:25: warning: excess elements in scalar initializer
  170 |     .close            = ff_vvenc_encode_close,
      |                         ^~~~~~~~~~~~~~~~~~~~~
libavcodec/libvvenc.c:170:25: note: (near initialization for 'ff_libvvenc_encoder')
libavcodec/libvvenc.c:171:5: error: field name not in record or union initializer
  171 |     .p.pix_fmts       = (const enum AVPixelFormat[]){ AV_PIX_FMT_YUV420P,
      |     ^
libavcodec/libvvenc.c:171:5: note: (near initialization for 'ff_libvvenc_encoder')
libavcodec/libvvenc.c:171:25: warning: excess elements in scalar initializer
  171 |     .p.pix_fmts       = (const enum AVPixelFormat[]){ AV_PIX_FMT_YUV420P,
      |                         ^
libavcodec/libvvenc.c:171:25: note: (near initialization for 'ff_libvvenc_encoder')
libavcodec/libvvenc.c:174:5: error: field name not in record or union initializer
  174 |     .p.capabilities   = AV_CODEC_CAP_DELAY | AV_CODEC_CAP_OTHER_THREADS,
      |     ^
libavcodec/libvvenc.c:174:5: note: (near initialization for 'ff_libvvenc_encoder')
In file included from libavcodec/avcodec.h:40,
                 from libavcodec/libvvenc.c:6:
libavcodec/codec.h:82:42: warning: excess elements in scalar initializer
   82 | #define AV_CODEC_CAP_DELAY               (1 <<  5)
      |                                          ^
libavcodec/libvvenc.c:174:25: note: in expansion of macro 'AV_CODEC_CAP_DELAY'
  174 |     .p.capabilities   = AV_CODEC_CAP_DELAY | AV_CODEC_CAP_OTHER_THREADS,
      |                         ^~~~~~~~~~~~~~~~~~
libavcodec/codec.h:82:42: note: (near initialization for 'ff_libvvenc_encoder')
   82 | #define AV_CODEC_CAP_DELAY               (1 <<  5)
      |                                          ^
libavcodec/libvvenc.c:174:25: note: in expansion of macro 'AV_CODEC_CAP_DELAY'
  174 |     .p.capabilities   = AV_CODEC_CAP_DELAY | AV_CODEC_CAP_OTHER_THREADS,
      |                         ^~~~~~~~~~~~~~~~~~
libavcodec/libvvenc.c:175:5: error: field name not in record or union initializer
  175 |     .caps_internal    = FF_CODEC_CAP_AUTO_THREADS | FF_CODEC_CAP_INIT_CLEANUP,
      |     ^
libavcodec/libvvenc.c:175:5: note: (near initialization for 'ff_libvvenc_encoder')
libavcodec/libvvenc.c:175:25: error: 'FF_CODEC_CAP_AUTO_THREADS' undeclared here (not in a function); did you mean 'AV_CODEC_CAP_AUTO_THREADS'?
  175 |     .caps_internal    = FF_CODEC_CAP_AUTO_THREADS | FF_CODEC_CAP_INIT_CLEANUP,
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~
      |                         AV_CODEC_CAP_AUTO_THREADS
libavcodec/libvvenc.c:175:53: error: 'FF_CODEC_CAP_INIT_CLEANUP' undeclared here (not in a function)
  175 |     .caps_internal    = FF_CODEC_CAP_AUTO_THREADS | FF_CODEC_CAP_INIT_CLEANUP,
      |                                                     ^~~~~~~~~~~~~~~~~~~~~~~~~
libavcodec/libvvenc.c:175:25: warning: excess elements in scalar initializer
  175 |     .caps_internal    = FF_CODEC_CAP_AUTO_THREADS | FF_CODEC_CAP_INIT_CLEANUP,
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~
libavcodec/libvvenc.c:175:25: note: (near initialization for 'ff_libvvenc_encoder')
libavcodec/libvvenc.c:176:5: error: field name not in record or union initializer
  176 |     .p.wrapper_name   = "libvvenc",
      |     ^
libavcodec/libvvenc.c:176:5: note: (near initialization for 'ff_libvvenc_encoder')
libavcodec/libvvenc.c:176:25: warning: excess elements in scalar initializer
  176 |     .p.wrapper_name   = "libvvenc",
      |                         ^~~~~~~~~~
libavcodec/libvvenc.c:176:25: note: (near initialization for 'ff_libvvenc_encoder')
cc1.exe: some warnings being treated as errors
make: *** [ffbuild/common.mak:81: libavcodec/libvvenc.o] Error 1

```

**Comment by 1480c1** [comment]
Seems it may need a few changes for the ffmpeg plugin, I can try to look into that once I have more time, but I would suggest first going through with adding vvenc and vvdec as executables first, so more people can test it and see if there are any other issues

**Comment by MartinEesmaa** [comment]
@1480c1, I think we can let anyone try build vvenc & vvdec after you merge my commits into this repository, so I'm ready to merge my commits by you. Sounds good idea!

- Martin Eesmaa

**Comment by 1480c1** [comment]
Sounds good, but do you want me to cleanup the commits, rebase and leave them as is, or squash them into just one commit?

**Comment by MartinEesmaa** [comment]
might be best to squash :)

**Comment by 1480c1** [comment]
@MartinEesmaa Thanks for the work!

**Comment by MartinEesmaa** [comment]
No problem, @1480c1! :)
- Martin Eesmaa

**Pull request merged**